### PR TITLE
Fix FB 1143542 - Slot names give better error messages

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
Nodes that accept custom text for inputs now validate for special characters that prevent the shader from compiling. 
SlotNameErrorMsg

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.
Validation check happens in `NodeUtils` and is called from nodes that need it.
Characters are valid for the display name are defined as A-Z, a-z, 0-9, _ ( ) and whitespace. Separators are for readability in the graph and get stripped before injected into the HLSL, and the validation will account for any non-Latin character edge cases now. 

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low

**Halo Effect**: None, Low, Medium, High?
Low

---
### Comments to reviewers
Notes for the reviewers you have assigned.
bug: https://fogbugz.unity3d.com/f/cases/1143542/

### CODE REVIEW

**Comment by zadi0648** [comment]
Adding Martin because of the reason described above

**Comment by martint-unity** [comment]
So, yes I have a PR that fixes it starting with digits. Also I don't think it is the right solution to do a string with characters to check. I guess regex should be used. Otherwise you will have to check all different characters like my native language Swedish åäö should also be added to that string I guess and many more crazy native keyboard chars (double ss in German etc). Just my 2 cents. I will add the regex as a fix when I am back if it hasn't been added by then.

**Comment by pbbastian** [comment]
Maybe not regex, but I think it should be turned on its head, so that there's a list of allowed characters, and then we check if it contains anything not in that list.

**Comment by alindmanUnity** [comment]
Good points, I can remove the digit check if that's no longer an error we need to worry about and flip the invalid character check.

File: com.unity.shadergraph/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - When you rapidly perform an undo or redo, Shader Graph windows no longer break.
 - Sub Graphs that contain references to non-existing Sub Graphs no longer break the Sub Graph Importer.
 - You can now reference sub-assets such as Textures.
+- SubGraph Output and CustomFunctionNode now validate slot names and display appropriate error badge.
```
Comment (sharlenet) [inline com.unity.shadergraph/CHANGELOG.md]: Sub Graph Output and Custom Function Node now validate slot names, and display appropriate error badge when needed.

**Comment by alindmanUnity** [comment]
@zadi0648 HLSL won't compile with those special characters, so it wouldn't benefit anyone to let them use it in the graph.
